.travis.yml

```diff
@@ -1,47 +1,72 @@
+  - CXXFLAGS="-std=c++11"
 
-cache: apt
+cache:
+  apt: true
+  directories:
+    - $HOME/.cache/pip
+    - $TRAVIS_BUILD_DIR/../ct
+    - $TRAVIS_BUILD_DIR/../depot_tools
+    - $TRAVIS_BUILD_DIR/../install
+
+addons:
+  apt:
+    sources:
+      - ubuntu-toolchain-r-test
+      - llvm-toolchain-precise-3.6
+    packages:
+    - autoconf
+    - automake
+    - build-essential
+    - clang-3.6
+    - python-dev
+    - tcl8.5
+    - libstdc++-4.9-dev
 
 before_install:
- - sudo apt-add-repository -y ppa:sao/backports
- - sudo apt-add-repository -y ppa:ubuntu-toolchain-r/test
- - sudo apt-add-repository -y ppa:fcitx-team/nightly
- - sudo apt-add-repository -y ppa:chris-lea/protobuf
- - sudo apt-add-repository -y ppa:asolovets/backports
- - sudo apt-get update -qq
- - sudo apt-get install -qq openssl libssl-dev autoconf automake protobuf-compiler libprotobuf-java libprotobuf-dev python-dev libjson-c-dev libgoogle-glog-dev libgflags-dev libldns-dev libstdc++-4.8-dev libleveldb-dev libsnappy-dev libgoogle-perftools-dev
-# Stupid frikkin' google-mock package on Precise is b0rked, so hack it up:
- - wget https://googlemock.googlecode.com/files/gmock-1.7.0.zip -O /tmp/gmock-1.7.0.zip
- - unzip -d /tmp /tmp/gmock-1.7.0.zip
- - sudo mv /tmp/gmock-1.7.0 /usr/src/gmock
- - wget https://github.com/ellzey/libevhtp/archive/1.2.10.zip -O /tmp/libevhtp-1.2.10.zip
- - unzip -d /tmp/ /tmp/libevhtp-1.2.10.zip
- - mkdir /tmp/libevhtp-1.2.10/build
- - pushd /tmp/libevhtp-1.2.10/build && cmake -DEVHTP_DISABLE_REGEX:STRING=ON -DCMAKE_C_FLAGS="-fPIC" .. && sudo make install; popd
- - sudo pip install -r python/requirements.txt
- - sudo ln -s protobuf-java.jar /usr/share/java/protobuf.jar
+ - if [ "$CXX" == "clang++" ] && which clang++-3.6; then export CXX="clang++-3.6" CC="clang-3.6"; fi
+ - if [ "$CXX" == "g++" ] && which g++-4.8; then export CXX="g++-4.8" CC="gcc-4.8"; fi
+ - if [ "$TRAVIS_OS_NAME" == "osx" ]; then . ./.travis.osx.before_install.sh; fi
+ - export INSTALL_DIR=$TRAVIS_BUILD_DIR/../install
+ - mkdir -p ${INSTALL_DIR}/include ${INSTALL_DIR}/lib ${INSTALL_DIR}/java
+ - echo "INSTALL_DIR=${INSTALL_DIR}"
+ - export PROTOC="${INSTALL_DIR}/bin/protoc"
+ - export GOPATH=${INSTALL_DIR}/go
  - mkdir -p $GOPATH/src/github.com/google
- - ln -s $PWD $GOPATH/src/github.com/google
+ - ln -sf $PWD $GOPATH/src/github.com/google
+ - echo "jars.protobuf = ${INSTALL_DIR}/java/protobuf.jar" > local.properties
+ - echo "protoc.bin = ${PROTOC}" >> local.properties
+
+install:
+ - autoconf --version
+ - pushd ..
+ - mkdir -p ct
+ - wget http://central.maven.org/maven2/com/google/protobuf/protobuf-java/2.6.0/protobuf-java-2.6.0.jar -O ${INSTALL_DIR}/java/protobuf.jar
+ - if [ ! -e depot_tools/gclient ]; then git clone https://chromium.googlesource.com/chromium/tools/depot_tools.git; fi
+ - export PATH="$PATH:/usr/local/bin:"`pwd`/depot_tools
+ - cd ct
+ - ln -s ../certificate-transparency certificate-transparency
```

Comment:
Well, `gclient` needs to know what it's going to be checking out, and if we don't supply an "unmanaged" `certificate-transparency` directory containing the `DEPS` file, then it'll helpfully check out the repo spec'd by the `gclient config` line below, which, obviously, won't be the particular repo/branch that we're actually trying to test.

I spent some time, with @daviddrysdale's help, trying to coax gclient to check out the correct thing for `certificate-transparency/` to no avail, so we symlink it in, which works just fine.